samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -173,152 +247,191 @@ class DailyCallManager {
-   */
-  playAudio(event) {
-    if (event.participant.local) {
-      return; // Avoid playing local audio
+    switch (track.state) {
+      case 'playable':
+        videoEl.style.display = '';
+        break;
+      case 'off':
+      case 'interrupted':
+      case 'blocked':
+        videoEl.style.display = 'none'; // Hide video but keep container
+        break;
+      default:
+        // Here we handle all other states the same as we handle 'playable'.
+        // In your code, you may choose to handle them differently.
+        videoEl.style.display = '';
+        break;
     }
-    const audioEl = document.createElement('audio'); // Create <audio> element
-    document.body.appendChild(audioEl); // Add <audio> element to DOM
-    audioEl.srcObject = new MediaStream([event.track]); // Attach audio track to <audio> element
-    audioEl.play(); // Play audio track
   }
 
   /**
-   * Removes a media element (audio or video) from the DOM for a specific participant
-   * and ensures media resources are released by setting srcObject to null.
-   * @param {string} participantId - The ID of the participant.
-   * @param {string} trackKind - The kind of track ('video' or 'audio').
+   *
+   * Creates or updates the video element for a participant to display their video stream.
+   * It either:
+   * - attaches a new MediaStreamTrack to the video element and plays it OR
+   * - checks if the video element already exists with a persistentTrack
+   *
+   * If the existing persistentTrack is different from the new one, it updates the srcObject
+   * with the new track and plays it. This ensures that the video element is efficiently managed,
+   * without unnecessary re-attachments or plays when the track has not changed.
+   *
+   * This framework is critical for handling dynamic video situations, such as when participants
+   * toggle their camera on or off, or when there is a track change due to some action within the call
+   * environment (e.g., switching devices).
+   *
+   * @param {Object} track - The video track data from the participant. This includes the actual
+   * MediaStreamTrack (`persistentTrack`) to be displayed in the video element.
+   * @param {string} participantId - The unique identifier for the participant whose video
+   * track is being handled. This ID is used to name and fetch specific video elements related
+   * to the participant in the DOM.
    */
-  removeMediaElement(participantId, trackKind) {
-    // Construct the selector based on the track kind
-    let elementSelector;
-    if (trackKind === 'video') {
-      elementSelector = `#video-container-${participantId} video`;
-    } else if (trackKind === 'audio') {
-      elementSelector = 'audio';
-    } else {
-      console.error('Unsupported track kind:', trackKind);
+  startOrUpdateVideo(track, participantId) {
+    const videoContainer = document.getElementById(
+      `video-container-${participantId}`
+    );
+    if (!videoContainer) {
```

Comment:
mostly academic/code-smell feedback, so feel free to ignore, but:

the fact that the video container is created in `updateVideoUi()` means that there is this dependency between `updateVideoUi()` and `startOrUpdateVideo()`—the order in which they're invoked within `handleParticipantJoinedOrUpdated()` matters. they're otherwise independent.

i think the better place to do the video container setup would be directly in `handleParticipantJoinedOrUpdated()`. another sign that the code is asking for it to be there is that the tear-down of the container occurs in `handleParticipantLeft()`.

again, fairly minor, feel free to ignore if you want.